pilot/pkg/security/authz/builder/extauthz.go

```diff
@@ -101,6 +87,36 @@ func buildExtAuthz(configs []*meshconfig.MeshConfig_ExtensionProvider, providers
 		}
 		resolved[config.Name] = parsed
 	}
+	if errs != nil {
+		return nil, errs
+	}
+
+	if authzLog.DebugEnabled() {
+		authzLog.Debugf("Resolved extension providers: %v", spew.Sdump(resolved))
+	}
+	return resolved, nil
+}
+
+func notAllTheSame(names []string) bool {
+	for i := 1; i < len(names); i++ {
+		if names[i-1] != names[i] {
+			return true
+		}
+	}
+	return false
+}
+
+func getExtAuthz(resolved map[string]*builtExtAuthz, providers []string) (*builtExtAuthz, error) {
+	if notAllTheSame(providers) {
+		return nil, fmt.Errorf("all extension providers must be the same for a specific workload, found multiple different providers: %v", providers)
+	} else if len(providers) < 1 {
```

Comment:
All the conditions must be not triggered to get a valid input right?

```
if resolved == nil {
  return;
}
if len(proviers) < 1 { return }
if notAlltheSame) { return }
```
would make it clearer as well the computation on the notAllSame can be avoided in advance.

Reply:
sounds good, updated.